### DISCUSSION TITLE
Understanding "Stop charging when sleeping" in clamshell mode

### DISCUSSION THREAD

**Discussion by jlpajares** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#1362] [Q&A]
Hola! I can't enable "Stop charging when sleeping". The switch is disabled. I'm not sure if this is a bug or an expected behaviour, since [this help page](https://apphousekitchen.com/when-does-aldente-disable-sleep-on-macbooks/) is mentioning "technical constraints" when the Macbook is in clamshell mode with the screen connected, as I have mine.
![CleanShot 2024-11-02 at 18 44 39@2x](https://github.com/user-attachments/assets/e9c986ab-df6b-45b3-bb3b-cb66dcada8f8)

**Comment by MatthiasKerbl** [comment]
Hi @jlpajares,

You probably have the feature "Disable Sleep Until Charge Limit" enabled. While this feature is enabled, "Stop charging when sleeping" is greyed out. This is because the scenario in which "Stop charging when sleeping" would come into action does not happen due to "Disable Sleep Until Charge Limit".

Best,
Matthias

**Reply by jlpajares** [reply]
Thanks @MatthiasKerbl! You're spot on, I did have "Disable Sleep Until Charge Limit" enabled. When I turned it off, the switch for "Disable Sleep Until Charge Limit" became available. Funny thing is, even with it disabled, I've noticed my MacBook's been out of juice in the mornings since I installed AlDente a couple of days ago. On the bright side, I guess my MacBook didn't mind this unintentional calibration process. I'll keep an eye on what happens now that I've activated "Disable Sleep Until Charge Limit".

Quick question: Should I maybe avoid using the MacBook at home in clamshell mode to maximize battery life? Appreciate any advice!

**Reply by MatthiasKerbl** [reply]
> Funny thing is, even with it disabled, I've noticed my MacBook's been out of juice in the mornings since I installed AlDente a couple of days ago.

This should not be the case. If possible, please share a debug file the next time this happens so that we can assess this issue better. You can find a guide on [How to generate and share a debug file](https://apphousekitchen.com/how-to-generate-and-share-a-debug-file/) on our blog.

> Quick question: Should I maybe avoid using the MacBook at home in clamshell mode to maximize battery life? Appreciate any advice!

In general, using the MacBook in clamshell mode is not an issue for battery longevity. It always depends on the specific usage scenario and setup.

Best,
Matthias
